### ISSUE TITLE
Connection configuration does not properly override with mixed definitions

### ISSUE THREAD

**Issue by mikary** [dotnet/efcore#1390]
The following program should use the connection string from the last definition loaded in the configuration (config.ini) but uses the definition from the first (config.json). Same inverted behavior is observed if the order of the AddXxxFile calls is reversed.

``` cs
    public class MyContext : DbContext
    {
        public MyContext(IServiceProvider serviceProvider)
            : base(serviceProvider)
        {
        }
    }
```

``` cs
    public class Program
    {
        public void Main(string[] args)
        {
            var configuration = new Configuration();
            configuration.AddJsonFile("config.json");
            configuration.AddIniFile("config.ini");

            var serviceProvider = new ServiceCollection()
                .AddEntityFramework(configuration)
                .AddSqlServer()
                .AddDbContext<MyContext>()
                .ServiceCollection
                .BuildServiceProvider();

            using (var context = new MyContext(serviceProvider))
            {
                context.Database.EnsureCreated();
                Console.WriteLine(context.Database.AsRelational().Connection.ConnectionString);
                Console.ReadLine();
            }
        }
    }
```

config.json:

``` json
{
    "Data": {
        "DefaultConnection": {
            "ConnectionString": "Data Source=(localdb)\\v11.0;Initial Catalog=MyDBJson;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30"
        }
    },

    "EntityFramework": {
        "MyContext": {
            "ConnectionStringKey": "Data:DefaultConnection:ConnectionString",
        }
    }
}
```

config.ini

``` ini
[EntityFramework]
MyContext:ConnectionString=Data Source=(localdb)\v11.0;Initial Catalog=MyDBIni;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30
```

**Comment by smitpatel** [comment]
The subkeys from configuration are generated in reverse order of the sources added. So in this case json file is added first so "ConnectionStringKey" comes last and stores the value from JSON file. (As mentioned in #1391 last one wins). Inverting order of files changes the order of subkeys and hence value comes from INI file.

**Comment by rowanmiller** [comment]
This will be resolved by the changes we're making tracked by #1391
